README
NAME
    Data::Password::Common - Check a password against a list of common
    passwords

VERSION
    version 0.003

SYNOPSIS
      use Data::Password::Common 'found';

      if ( found( $password ) ) {
        die "'$password' is a common password"
      }

      # import with aliasing
      use Data::Password::Common found => { -as => "found_common" };

      # custom common password list
      use Data::Password::Common found => { list => "/usr/share/dict/words" };

DESCRIPTION
    This module installs a list of over 557,000 common passwords and
    provides a function to check a string against the list.

    The password list is taken from InfoSecDaily at
    <http://www.isdpodcast.com/resources/62k-common-passwords/>. (They claim
    their list is over 62K, but they must have misread their "wc" output.)

USAGE
    Functions are provided via Sub::Exporter. Nothing is exported by
    default.

  found
      found($password);

    Returns true if the password is in the common passwords list.

CUSTOMIZING
    You may choose an alternate password list to check by passing a "list"
    parameter during import:

      use Data::Password::Common found => { list => "/usr/share/dict/words" };

    The file must be sorted.

SEE ALSO
  Password checkers
    *   Data::Password

    *   Data::Password::Entropy

    *   Data::Password::BasicCheck

  Lists of common passwords
    *   InfoSecDaily
        <http://www.isdpodcast.com/resources/62k-common-passwords/>

    *   Skull Security
        <http://www.skullsecurity.org/wiki/index.php/Passwords>

SUPPORT
  Bugs / Feature Requests
    Please report any bugs or feature requests through the issue tracker at
    <https://github.com/dagolden/Data-Password-Common/issues>. You will be
    notified automatically of any progress on your issue.

  Source Code
    This is open source software. The code repository is available for
    public review and contribution under the terms of the license.

    <https://github.com/dagolden/Data-Password-Common>

      git clone https://github.com/dagolden/Data-Password-Common.git

AUTHOR
    David Golden <dagolden@cpan.org>

COPYRIGHT AND LICENSE
    This software is Copyright (c) 2012 by David Golden.

    This is free software, licensed under:

      The Apache License, Version 2.0, January 2004



Changes
Revision history for Data-Password-Common

0.003     2013-10-14 06:46:05 America/New_York

    [FIXED]

    - Deduplicated password list and re-encoded it as UTF-8

0.002     2013-01-02 12:57:01 America/New_York

    [ADDED]

    - Allows specifying a custom password list

    [DOCUMENTED]

    - Clarified that the password list is actually over 620,000

0.001     2012-12-05 12:58:56 America/New_York

    - First release



Makefile.PL

use strict;
use warnings;

use 5.010000;

use ExtUtils::MakeMaker 6.17;

use File::ShareDir::Install;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "Check a password against a list of common passwords",
  "AUTHOR" => "David Golden <dagolden\@cpan.org>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.17",
    "File::ShareDir::Install" => "0.03"
  },
  "DISTNAME" => "Data-Password-Common",
  "EXE_FILES" => [],
  "LICENSE" => "apache",
  "NAME" => "Data::Password::Common",
  "PREREQ_PM" => {
    "File::ShareDir" => 0,
    "IO::File" => 0,
    "Search::Dict" => 0,
    "Sub::Exporter" => 0,
    "autodie" => "2.00",
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec::Functions" => 0,
    "List::Util" => 0,
    "Test::File::ShareDir" => 0,
    "Test::More" => "0.96",
    "utf8" => 0
  },
  "VERSION" => "0.003",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  my $tr = delete $WriteMakefileArgs{TEST_REQUIRES};
  my $br = $WriteMakefileArgs{BUILD_REQUIRES};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}




lib/Data/Password/Common.pm
use v5.10;
use strict;
use warnings;

package Data::Password::Common;
# ABSTRACT: Check a password against a list of common passwords
our $VERSION = '0.003'; # VERSION

# Dependencies
use File::ShareDir;
use IO::File;
use Search::Dict;
use autodie 2.00;

use Sub::Exporter -setup => { exports => [ 'found' => \&build_finder ] };

sub build_finder {
    my ( $class, $name, $arg, $col ) = @_;
    my $list_path = $arg->{list}
      || File::ShareDir::dist_file( "Data-Password-Common", "common.txt" );
    my $list_handle = IO::File->new( $list_path, "<:utf8" );

    return sub {
        return unless @_;
        my $password = shift;
        look $list_handle, $password;
        chomp( my $found = <$list_handle> );
        return $found eq $password;
    };
}

1;


# vim: ts=2 sts=2 sw=2 et:

__END__

=pod

=encoding utf-8

=head1 NAME

Data::Password::Common - Check a password against a list of common passwords

=head1 VERSION

version 0.003

=head1 SYNOPSIS

  use Data::Password::Common 'found';

  if ( found( $password ) ) {
    die "'$password' is a common password"
  }

  # import with aliasing
  use Data::Password::Common found => { -as => "found_common" };

  # custom common password list
  use Data::Password::Common found => { list => "/usr/share/dict/words" };

=head1 DESCRIPTION

This module installs a list of over 557,000 common passwords and provides
a function to check a string against the list.

The password list is taken from InfoSecDaily at
L<http://www.isdpodcast.com/resources/62k-common-passwords/>. (They claim their
list is over 62K, but they must have misread their C<wc> output.)

=for Pod::Coverage build_finder

=head1 USAGE

Functions are provided via L<Sub::Exporter>.  Nothing is exported by default.

=head2 found

  found($password);

Returns true if the password is in the common passwords list.

=head1 CUSTOMIZING

You may choose an alternate password list to check by passing a C<list> parameter
during import:

  use Data::Password::Common found => { list => "/usr/share/dict/words" };

The file must be sorted.

=head1 SEE ALSO

=head2 Password checkers

=over 4

=item *

L<Data::Password>

=item *

L<Data::Password::Entropy>

=item *

L<Data::Password::BasicCheck>

=back

=head2 Lists of common passwords

=over 4

=item *

L<InfoSecDaily|http://www.isdpodcast.com/resources/62k-common-passwords/>

=item *

L<Skull Security|http://www.skullsecurity.org/wiki/index.php/Passwords>

=back

=for :stopwords cpan testmatrix url annocpan anno bugtracker rt cpants kwalitee diff irc mailto metadata placeholders metacpan

=head1 SUPPORT

=head2 Bugs / Feature Requests

Please report any bugs or feature requests through the issue tracker
at L<https://github.com/dagolden/Data-Password-Common/issues>.
You will be notified automatically of any progress on your issue.

=head2 Source Code

This is open source software.  The code repository is available for
public review and contribution under the terms of the license.

L<https://github.com/dagolden/Data-Password-Common>

  git clone https://github.com/dagolden/Data-Password-Common.git

=head1 AUTHOR

David Golden <dagolden@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is Copyright (c) 2012 by David Golden.

This is free software, licensed under:

  The Apache License, Version 2.0, January 2004

=cut
